sky/clouds/aws.py

```diff
@@ -400,15 +401,19 @@ def make_deploy_resources_variables(self,
 
         user_security_group = skypilot_config.get_nested(
             ('aws', 'security_group_name'), None)
-        if resources.ports is not None:
+        if user_security_group is not None and not isinstance(
+                user_security_group, str):
+            for profile in user_security_group:
+                if fnmatch.fnmatchcase(cluster_name_on_cloud,
```

Comment:
This is a good catch! A proposal would be we refactor the `make_deploy_resources_variable` to take `ClusterName` dataclass in all cloud classes, instead of a single string for the cluster name:
https://github.com/skypilot-org/skypilot/blob/05aafb8b5970acb353eac24c4ee7ef0c6a97e118/sky/provision/provisioner.py#L41-L50

To do so, we can move the `ClusterName` class out of `provision` module to `utils.resources_utils`, and let the cloud class and the functions under `provision` module to use `ClusterName` object as input.